link/src/main/java/com/stripe/android/link/repositories/LinkApiRepository.kt

```diff
@@ -60,15 +60,18 @@ internal class LinkApiRepository @Inject constructor(
         name: String?,
         consentAction: ConsumerSignUpConsentAction
     ): Result<ConsumerSessionSignup> = withContext(workContext) {
-        stripeRepository.consumerSignUp(
-            email = email,
-            phoneNumber = phone,
-            country = country,
-            name = name,
-            locale = locale,
-            consentAction = consentAction,
-            requestOptions = buildRequestOptions(),
-        )
+        runCatching {
```

Comment:
rather than adding a `runCatching` here, can you update the new `signUp` endpoint to return a `Result`? And if you want to throw in FC, you can just call `getOrThrow`.

Reply:
Just went with the prevalent approach in `ConsumersApiService`, which is to throw. But you’re right: we should have repositories return `Result<T>` instead of throwing.

Reply:
great, thanks for the update!